drivers/flash/flash_stm32_ospi.c

```diff
@@ -1159,10 +1159,7 @@ static int flash_stm32_ospi_init(const struct device *dev)
 		return ret;
 	}
 
-	/* Initializes the independent peripherals clock */
-	__HAL_RCC_OSPI_CONFIG(RCC_OSPICLKSOURCE_SYSCLK); /* */
-
-	/* Clock configuration */
+	/* Clock configuration, including the alternate clock for peripheral */
 	if (clock_control_on(DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE),
 			     (clock_control_subsys_t) &dev_cfg->pclken) != 0) {
```

Comment:
You miss to add some code that will query the additional clock cells. Cf https://github.com/zephyrproject-rtos/zephyr/pull/45508 for instance.

Reply:
done